pkg/shared/util/string_test.go

```diff
@@ -20,6 +20,7 @@ import (
 	"strings"
 	"testing"
 
+	"github.com/numaproj/numaflow/pkg/apis/numaflow/v1alpha1"
```

Comment:
This is a bit of a nitpicking but I feel it's important from a code structure perspective - string utils shall not have knowledge of upper level logics like `v1alpha1`.

This limits `CompareSlice` in a way that it can only be used by `edge_types.go` while in fact it can be a util function that can be used everywhere.

I would suggest updating it to `func CompareSlice(operator string, sa []string, sb []string) bool` and switch on the string value instead of `v1alpha1.LogicOperator`.

Reply:
+1, we might unknowingly end up in a cyclic-dependency

Reply:
lets keep it as is, we can change it later.